src/silx/io/url.py

```diff
@@ -197,60 +228,41 @@ def __check_validity(self):
         elif self.__scheme == "silx":
             # If there is a slice you must have a data path
             # But you can have a data path without slice
-            slice_implies_data = (self.__data_path is None and self.__data_slice is None) or self.__data_path is not None
+            slice_implies_data = (
+                self.__data_path is None and self.__data_slice is None
+            ) or self.__data_path is not None
             self.__is_valid = slice_implies_data
         else:
             self.__is_valid = False
 
     @staticmethod
-    def _parse_slice(slice_string):
+    def _parse_slice(slice_string: str) -> Tuple[SliceLike, ...]:
         """Parse a slicing sequence and return an associated tuple.
 
         It supports a sequence of `...`, `:`, and integers separated by a coma.
-
-        :rtype: tuple
         """
-        def str_to_slice(string):
-            if string == "...":
-                return Ellipsis
-            elif ':' in string:
-                if string == ":":
-                    return slice(None)
-                else:
-                    def get_value(my_str):
-                        if my_str in ('', None):
-                            return None
-                        else:
-                            return int(my_str)
-                    sss = string.split(':')
-                    start = get_value(sss[0])
-                    stop = get_value(sss[1] if len(sss) > 1 else None)
-                    step = get_value(sss[2] if len(sss) > 2 else None)
-                    return slice(start, stop, step)
-            else:
-                return int(string)
```

Comment:
Moved out of the method and renamed `string_to_slice` for consistency with other functions that do the reverse operation.